pyvista/core/utilities/transform.py

```diff
@@ -0,0 +1,676 @@
+"""Module containing the Transform class."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from typing import Literal
+
+from pyvista.core import _validation
+from pyvista.core import _vtk_core as _vtk
+from pyvista.core.utilities.arrays import array_from_vtkmatrix
+from pyvista.core.utilities.arrays import vtkmatrix_from_array
+
+if TYPE_CHECKING:  # pragma: no cover
+    from pyvista.core._typing_core import NumpyArray
+    from pyvista.core._typing_core import RotationLike
+    from pyvista.core._typing_core import TransformLike
+
+
+class Transform(_vtk.vtkTransform):
```

Comment:
Thanks for adding great class! It is possible to make the operation of the Transform class even more intuitive to use with ["operator — Standard operators as functions"](https://docs.python.org/3/library/operator.html#operator) in my mind :)

Reply:
Great idea! Hadn't thought of that. I was thinking this PR is already a bit big with approx 800-900 LOC, maybe we can add that as a separate PR?

This class can also be extended to incorporate `reflect`, `rotate_x`, etc. transformations. And I was thinking of adding a `point` kwarg too for rotation about a point, reflection about a point, etc. 

Lot's of possible features to add.

Reply:
> Thanks for adding great class! It is possible to make the operation of the Transform class even more intuitive to use with ["operator — Standard operators as functions"](https://docs.python.org/3/library/operator.html#operator) in my mind :)

To add to this, we could even make this class comply with numpy's array protocols, such that you can do `np.array(transform)` or `np.eye(4)` @ `transform` which should return a numpy array. Or even implement `__getitem__` to do `element = transform[0,1]`.

As an implementation note (for future PRs), this means that the `matrix_list` should return a list of `Transform` instances instead of currently doing `array_from_vtkmatrix(transform.GetMatrix())`